Cargo.toml

```diff
@@ -25,6 +26,7 @@ seq-macro = "0.3"
 enable_logging = ["env_logger", "log"]
 stream = []
 raw_decoder = []
+no_std = ["core2"]
```

Comment:
This is 100% opposite of what is suggested in the [cargo book](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification):

"features should be additive. That is, enabling a feature should not disable functionality, and it should usually be safe to enable any combination of features. A feature should not introduce a [SemVer-incompatible change](https://doc.rust-lang.org/cargo/reference/features.html#semver-compatibility).

For example, if you want to optionally support [no_std](https://doc.rust-lang.org/reference/names/preludes.html#the-no_std-attribute) environments, do not use a no_std feature. Instead, use a std feature that enables std."

In my experience, people are definitely following the cargo book's recommendation. So I think you had it right the first time.

Reply:
Thanks for the review.

> Instead, use a std feature that enables std

I think this advice is for the general case, when optional dependencies aren't involved. It's not possible to use optional dependencies in a `no_std` build using the `std` feature approach.

This is one of the main reasons @gendx suggested using the `no_std` feature approach.

Reverting to the original PR isn't much work, but the changes do have their drawbacks.

> A feature should not introduce a [SemVer-incompatible change](https://doc.rust-lang.org/cargo/reference/features.html#semver-compatibility).

This PR (both versions) do not break SemVer, users would still get `std` builds by default. Compiled code should literally be identical to code without the changes, unless the opt-in feature is used.

In the original, the user opts-in with `--default-features false`. In the current version, the user opts-in with `--features no_std`. Otherwise, the build should use all the `std` dependencies, and be identical to the current mainline.

Reply:
`std` should be part of your defaults, people who want to use it with no_std will use `--default-features false` (or ` default-features = false` in the Cargo.toml). This is the pattern that exists across all the crates I work with in the embedded space.

Reply:
> `std` should be part of your defaults, people who want to use it with no_std will use `--default-features false` (or ` default-features = false` in the Cargo.toml). This is the pattern that exists across all the crates I work with in the embedded space.

I agree with you, that's why I did it that way in the original PR. The (now-)optional dependency gets dropped from compilation when `std` feature is enabled, since it's not used anywhere else. That essentially turns a required dependency into an optional dependency, with the exception that the crate (and its dependencies) are always downloaded.

I'm fine with either version, honestly. I can revert to the original changes, or keep the PR as it is now. It's up to what @gendx wants to do.

Still no movement on the `core2` PR.